src/app/common/modals/progress-modal/progress-modal.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+@Component({
+  selector: 'progress-modal',
+  templateUrl: './progress-modal.component.html',
+})
+export class ProgressModalComponent {
+  constructor(
+    public dialogRef: MatDialogRef<any>,
+    @Inject(MAT_DIALOG_DATA) public data: any
```

Comment:
Data Injection: Injecting data using @Inject(MAT_DIALOG_DATA) is fine, but make sure to provide a more specific type for the data if possible instead of using "any" to improve type safety.